example/newreposecret/main.go

```diff
@@ -0,0 +1,197 @@
+	secretValue := os.Getenv(secretName)
+	if secretValue == "" {
+		return "", errors.New(fmt.Sprintf(`secret with name: %s not defined as an environment variable, 
+please export the variable as an enviornment variable so it can be read in
+
+ex:
+	export %s="secret value"
+`, secretName, secretName))
+	}
+	return secretValue, nil
+}
+
+// GithubAuth reads an api token from the environment
+// expecting API_GITHUB_TOKEN and authenticates with github api
+// and returns an authenticated client.
+func GithubAuth() (context.Context, *github.Client, error) {
+	token := os.Getenv("API_GITHUB_TOKEN")
+	if token == "" {
+		return nil, nil, errors.New(`no API_GITHUB_TOKEN was found in the environment variables.
+This is needed to authenticate with github. Please generate a github token that has access to do what you are trying to do
+and export it as an environment variable.
+
+ex:
+	export API_GITHUB_TOKEN="<token contents string>"
+`)
+	}
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+	return ctx, client, nil
+}
+
+// AddRepoSecret will add a secret value to a given github repo for a given owner
+// It encrypts the secret using sodium before sending the secret to github api
+// therefore requires libsodium to be installed on the machine running this code
+// https://formulae.brew.sh/formula/libsodium
+// Github is very picky over formats things need to be to be sent, and not very descriptive in how
+// they will be sent to the user
+// To get a secret uploaded you need to get the public key of the repo that will be receiving the
+// secret. This key is used to encrypt the secret before transport on the sending side, and then decrypted by github.
+// Once you have the public key you need to encrypt the secret with sodiumlib before sending it.
+// The public key comes base64 encoded, and sodiumlib expects it to not be base64 encoded so you need
+// to decode the public key before using it.
+// Once the public key is decoded you need to convert the string secret into bytes.
+// once you have the public key decoded, and the secret string in bytes you can encrypt it using
+// sodium.CryptoBoxSeal
+// That will produce the correctly encrypted secret as bytes, but you need to then convert it to
+// a base64 encoded string. After doing that you can use that base64 encoded string as the encrypted value
+// to be part of the github.EncodedSecret type.
+// The name is the string (no encoding, or base64 needed) of the secret name that will appear in github secrets
+// then the KeyID will be the public key of the repo's ID, which is gettable from the public key's GetKeyID method.
+// Finally you can pass that object in and have it be created or updated in github.
+func AddRepoSecret(ctx context.Context, client *github.Client, owner string, repo string, secretName string, secretValue string) (string, error) {
```

Comment:
This method does not need exporting, so it will be `addRepoSecret`.  Also, there is no reason to return the passed-in secretName, so don't.

```suggestion
func addRepoSecret(ctx context.Context, client *github.Client, owner, repo, secretName, secretValue string) error {
```

Reply:
unexported and removed the return. good idea!